src/ir/auth_logic/ast.h

```diff
@@ -19,13 +19,15 @@
 #ifndef SRC_IR_AUTH_LOGIC_AST_H_
 #define SRC_IR_AUTH_LOGIC_AST_H_
 
+#include "src/ir/common/datalog_ir.h"
 #include <memory>
 #include <string>
 #include <variant>
 #include <vector>
 
 #include "absl/hash/hash.h"
 
+namespace common_ir = raksha::ir::common;
```

Comment:
Just FYI. You don't need this using declaration. Anywhere within `raksha::ir` namespace, you can simply use `common::`.  (If you rename the namespace as `raksha::ir:datalog` as I suggested, you will simply use `datalog::`.)